### PR TITLE
Run rails app:update for 5.2.4.2

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Chore

## Description

We updated to the newest version of Rails yesterday, so I though it's time to run a `rails app:update` and apply the various changes manually. Maybe at one point in the not too distant future we can try an upgrade to Rails 6?

## Related Tickets & Documents

n/a

## Added tests?

- [X] no, because they aren't needed

## Added to documentation?

- [X] no documentation needed

## Are there any post deployment tasks we need to perform?

Make sure nothing explodes.

### CODE REVIEW

File: bin/update [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env ruby
```
Comment (rhymes) [inline bin/update]: we can remove this, see https://github.com/thepracticaldev/dev.to/pull/6566

File: config/application.rb [hunk 2]
```diff
@@ -1,38 +1,31 @@
...
-  sprockets/railtie
-].each do |railtie|
-  require railtie
-end
+require "rails"
+# Pick the frameworks you want:
+require "active_model/railtie"
+require "active_job/railtie"
+require "active_record/railtie"
+require "active_storage/engine"
```
Comment (rhymes) [inline config/application.rb]: we can comment this as we don't need it, yet :D less memory used in the meantime

File: config/environments/development.rb [hunk 3]
```diff
@@ -34,9 +35,14 @@
     config.cache_store = :null_store
   end
 
+  # Store uploaded files on the local file system (see config/storage.yml for options)
+  config.active_storage.service = :local
```
Comment (rhymes) [inline config/environments/development.rb]: let's comment this out as well

File: config/environments/production.rb [hunk 4]
```diff
@@ -38,17 +36,19 @@
...
-  # have moved to config/initializers/assets.rb
+  # Enable serving of images, stylesheets, and JavaScripts from an asset server.
+  # config.action_controller.asset_host = 'http://assets.example.com'
 
   # Specifies the header that your server uses for sending files.
   # config.action_dispatch.x_sendfile_header = 'X-Sendfile' # for Apache
   # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
 
+  # Store uploaded files on the local file system (see config/storage.yml for options)
+  config.active_storage.service = :local
```
Comment (rhymes) [inline config/environments/production.rb]: let's comment this out as well

File: config/environments/test.rb [hunk 5]
```diff
@@ -31,7 +31,7 @@
   config.action_controller.allow_forgery_protection = false
 
   # Store uploaded files on the local file system in a temporary directory
-  # config.active_storage.service = :test
+  config.active_storage.service = :test
```
Comment (rhymes) [inline config/environments/test.rb]: same as the other ones :D

**Comment by citizen428** [comment]
@rhymes Made all the changes now.

I think we should consider excluding files like the `application.rb` and the environment configurations from Rubocop via the `exclude` key. The diffs are huge, mostly because of things vs single and double quotes and our pre-commit hook failed with

```
config/environments/development.rb:23:6: C: Rails/FilePath: Please use Rails.root.join('path/to') instead.
if Rails.root.join("tmp", "caching-dev.txt").exist?
```

Consistency is nice, but maybe not at the cost of making upgrades harder.

**Comment by rhymes** [comment]
@citizen428 it makes sense! though we need to go back to single quotes there before then, right?

**Comment by citizen428** [comment]
@rhymes For now, I'd just keep it as is, but with the eventual update to Rails 6 I think we should switch it back and then ignore it in Rubocop.

File: config/environments/production.rb [hunk 6]
```diff
@@ -75,18 +80,23 @@
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
   # the I18n.default_locale when a translation cannot be found).
-  config.i18n.fallbacks = [I18n.default_locale]
+  config.i18n.fallbacks = true
```
Comment (rhymes) [inline config/environments/production.rb]: for anyone wondering about this, this is what <https://guides.rubyonrails.org/configuring.html> says:

> You can set the option to true for using default locale as fallback

**Pull request merged**